### PR TITLE
Support Web Redemption Links

### PR DESCRIPTION
**Pull request by tonidero**
This adds support for Web Redemption Links in our Flutter SDK.

### CODE REVIEW

File: lib/models/web_purchase_redemption_result.freezed.dart [hunk 1]
```diff
@@ -0,0 +1,875 @@
+// coverage:ignore-file
```
Comment (tonidero) [inline lib/models/web_purchase_redemption_result.freezed.dart:1]: This is generated code by freezed, no need to review it!

File: revenuecat_examples/purchase_tester/pubspec.yaml [hunk 2]
```diff
@@ -13,6 +13,8 @@ dependencies:
   # Use with the CupertinoIcons class for iOS style icons.
   cupertino_icons: ^1.0.4
 
+  app_links: ^6.3.3
```
Comment (tonidero) [inline revenuecat_examples/purchase_tester/pubspec.yaml:16]: I added this library to read the deep links. Seemed one of the simpler ones.

**Review by tonidero — COMMENTED** [review]
Sorry for the big diff, but more than half is actually generated code, so it should hopefully not be as bad as it looks 🙏

**Review by vegaro — APPROVED** [review]
looks good! I think this solution will just work fine with any deeplinking library

**Pull request merged**